### PR TITLE
Planning: openspace: parallel smoothing with partitioned hybrid astar trajectory

### PR DESCRIPTION
**Pull request by JasonZhou404**

### CODE REVIEW

File: modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc [hunk 1]
```diff
@@ -299,22 +296,33 @@ bool HybridAStar::GenerateSpeedAcceleration(HybridAStartResult* result) {
...
+
   // load velocity from position
-  for (size_t i = 0; i + 1 < x_size; ++i) {
-    const double discrete_v = ((result->x[i + 1] - result->x[i]) / delta_t_) *
-                                  std::cos(result->phi[i]) +
-                              ((result->y[i + 1] - result->y[i]) / delta_t_) *
-                                  std::sin(result->phi[i]);
+  // initial and end speed are set to be zeros
+  result->v.push_back(0.0);
+  for (size_t i = 1; i < x_size - 1; ++i) {
```
Comment (YajiaZhang) [inline modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc]: Better to write as i + 1 < x_size

File: modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc [hunk 2]
```diff
@@ -426,6 +434,98 @@ bool HybridAStar::GenerateSCurveSpeedAcceleration(HybridAStartResult* result) {
...
+  bool current_gear =
+      std::abs(common::math::NormalizeAngle(tracking_angle - heading_angle)) <
+      (M_PI / 2.0);
+  for (size_t i = 0; i < horizon - 1; ++i) {
+    heading_angle = phi[i];
+    const Vec2d tracking_vector(x[i + 1] - x[i], y[i + 1] - y[i]);
+    tracking_angle = tracking_vector.Angle();
+    bool gear =
+        std::abs(common::math::NormalizeAngle(tracking_angle - heading_angle)) <
+        (M_PI / 2.0);
```
Comment (YajiaZhang) [inline modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc]: You can use M_PI_2

File: modules/planning/conf/planning.conf [hunk 3]
```diff
@@ -30,3 +30,4 @@
 
 --enable_nonscenario_side_pass
 --noenable_smoother_failsafe
+--noenable_parallel_trajectory_smoothing
```
Comment (xiaoxq) [inline modules/planning/conf/planning.conf]: Clear red mark.

File: modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc [hunk 4]
```diff
@@ -426,6 +434,98 @@ bool HybridAStar::GenerateSCurveSpeedAcceleration(HybridAStartResult* result) {
...
+    current_traj->y.push_back(y[i]);
+    current_traj->phi.push_back(phi[i]);
+  }
+  current_traj->x.push_back(x.back());
+  current_traj->y.push_back(y.back());
+  current_traj->phi.push_back(phi.back());
+
+  // Retrieve v, a and steer from path
+  size_t traj_size = partitioned_result->size();
+  for (size_t i = 0; i < traj_size; ++i) {
```
Comment (xiaoxq) [inline modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc]: `for (auto& result : partitioned_result)`

File: modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc [hunk 5]
```diff
@@ -426,6 +434,98 @@ bool HybridAStar::GenerateSCurveSpeedAcceleration(HybridAStartResult* result) {
...
+
+bool HybridAStar::GetTemporalProfile(HybridAStartResult* result) {
+  std::vector<HybridAStartResult> partitioned_results;
+  if (!TrajectoryPartition(*result, &partitioned_results)) {
+    AERROR << "TrajectoryPartition fail";
+    return false;
+  }
+  HybridAStartResult stitched_result;
+  for (const auto& result : partitioned_results) {
+    std::copy(result.x.begin(), result.x.end() - 1,
```
Comment (xiaoxq) [inline modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc]: Comment that the `one off` is intended.

File: modules/tools/open_space_visualization/distance_approach_problem_wrapper.cc [hunk 6]
```diff
@@ -298,29 +293,183 @@ bool DistancePlan(HybridAStar* hybridA_ptr, ObstacleContainer* obstacles_ptr,
...
+                             &dual_n_result_ds_vec[i])) {
+        return false;
+      }
+    }
+
+    // Retrieve result in one single trajectory
+    size_t trajectory_point_size = 0;
+    for (size_t i = 0; i < size; ++i) {
+      trajectory_point_size +=
+          static_cast<size_t>(state_result_ds_vec[i].cols()) - 1;
```
Comment (xiaoxq) [inline modules/tools/open_space_visualization/distance_approach_problem_wrapper.cc]: Would `cols()` be 0? If yes, you'll get unexpected large integer. Make it safe.

**Pull request merged**